src/features/timebar/timebar.selectors.ts

```diff
@@ -15,75 +18,93 @@ type TimebarTrack = {
   color: string
 }
 
-export const getTracksData = createSelector(
-  [selectDataviews, selectTracks],
-  (dataviewWorkspaces, tracks) => {
-    const tracksSegments: TimebarTrack[] = []
-    Object.keys(tracks).forEach((id) => {
-      const dataviewWorkspace = dataviewWorkspaces.find(
-        (dataviewWorkspace: DataviewWorkspace) => id === dataviewWorkspace.datasetParams.id
+const selectTracksDataviews = createSelector([selectDataviews], (dataviewWorkspaces) => {
+  const dataviews: DataviewWorkspace[] = dataviewWorkspaces.filter(
+    (dataviewWorkspace: DataviewWorkspace) => {
+      return (
+        dataviewWorkspace.dataview?.config.type === Type.Track &&
+        dataviewWorkspace.dataview?.config.visible !== false
       )
-      if (!dataviewWorkspace || !dataviewWorkspace.dataview) return
-      const config = dataviewWorkspace.dataview.config
-      if (config.visible !== false) {
-        const trackSegments: TimebarTrackSegment[] = tracks[id].map((segment) => {
+    }
+  )
+  return dataviews
+})
+
+export const getTracksData = createSelector(
+  [selectTracksDataviews, selectTracks],
+  (trackDataviews, tracks) => {
+    const tracksSegments: TimebarTrack[] = trackDataviews.map(
+      (dataviewWorkspace: DataviewWorkspace) => {
+        const id = dataviewWorkspace.datasetParams.id
+        const track = tracks[id]
+        const trackSegments: TimebarTrackSegment[] = !track
+          ? []
+          : tracks[id].map((segment) => {
+              return {
+                start: segment[0].timestamp || 0,
+                end: segment[segment.length - 1].timestamp || 0,
+              }
+            })
+        return {
+          segments: trackSegments,
+          color: dataviewWorkspace.dataview?.config.color,
+        }
+      }
+    )
+
+    return tracksSegments
+  }
+)
+
+export const getTracksGraphs = createSelector(
+  [selectTracksDataviews, selectTracks, selectTimebarMode],
+  (trackDataviews, tracks, currentTimebarMode) => {
+    const graphs = trackDataviews.map((dataviewWorkspace: DataviewWorkspace) => {
+      const id = dataviewWorkspace.datasetParams.id
+      const trackSegments = tracks[id]
+      if (!trackSegments) return null
+      const color = dataviewWorkspace.dataview?.config.color
+      const segmentsWithCurrentFeature = trackSegments.map((segment) => {
+        return segment.map((pt) => {
+          const value = pt[currentTimebarMode as Field]
           return {
-            start: segment[0].timestamp || 0,
-            end: segment[segment.length - 1].timestamp || 0,
+            date: pt.timestamp,
+            value,
           }
         })
-        tracksSegments.push({
-          segments: trackSegments,
-          color: config.color,
-        })
+      })
+      return {
+        color,
+        segmentsWithCurrentFeature,
+        // TODO Figure out this magic value
+        maxValue: 20,
```

Comment:
🔮

Reply:
Where is this coming from? And why this value?

Reply:
The problem is that speed values returned by the API range from 0 to 1000+, when meaningful values should not really exceed 20/30 knots. Not sure yet what the best solution is:
- fix on the API side
- talk with Tim / Jaeyoon and establish a hardcoded value
- remove outliers mathematically (calculate for each value if they are within 1 standard deviation of mean?)

Reply:
- fixing it on the API means potentially breaking it for other use cases (unless is a url parameter)
- if we go the route of removing outliers based on standard deviation I'd choose 3 or 4 std to be safe
- hardcoded value seams fine for me (maybe change the comment to explain the domain knowledge behind the number?)